src/hosts.rs

```diff
@@ -17,25 +17,67 @@ pub enum HostStatus {
     OfflineHard,
 }
 
-impl fmt::Display for HostStatus {
+impl fmt::Display for ProxyStatus {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            HostStatus::Online => write!(f, "ONLINE"),
-            HostStatus::Shunned => write!(f, "SHUNNED"),
-            HostStatus::OfflineSoft => write!(f, "OFFLINE_SOFT"),
-            HostStatus::OfflineHard => write!(f, "OFFLINE_HARD"),
+            ProxyStatus::Online => write!(f, "ONLINE"),
+            ProxyStatus::Shunned => write!(f, "SHUNNED"),
+            ProxyStatus::OfflineSoft => write!(f, "OFFLINE_SOFT"),
+            ProxyStatus::OfflineHard => write!(f, "OFFLINE_HARD"),
         }
     }
 }
 
-impl From<String> for HostStatus {
+impl From<String> for ProxyStatus {
     fn from(s: String) -> Self {
         match s.to_uppercase().as_str() {
-            "ONLINE" => HostStatus::Online,
-            "SHUNNED" => HostStatus::Shunned,
-            "OFFLINE_SOFT" => HostStatus::OfflineSoft,
-            "OFFLINE_HARD" => HostStatus::OfflineHard,
-            _ => HostStatus::Online,
+            "ONLINE" => ProxyStatus::Online,
+            "SHUNNED" => ProxyStatus::Shunned,
+            "OFFLINE_SOFT" => ProxyStatus::OfflineSoft,
+            "OFFLINE_HARD" => ProxyStatus::OfflineHard,
+            _ => ProxyStatus::Online,
+        }
+    }
+}
+
+impl From<ReadysetStatus> for ProxyStatus {
+    fn from(status: ReadysetStatus) -> Self {
+        match status {
+            ReadysetStatus::Online => ProxyStatus::Online,
+            ReadysetStatus::SnapshotInProgress => ProxyStatus::Shunned,
+            ReadysetStatus::Maintenance => ProxyStatus::OfflineSoft,
+            ReadysetStatus::Unknown => ProxyStatus::Shunned,
+        }
+    }
+}
+
+#[derive(PartialEq, Clone, Copy)]
+
```

Comment:
nit: Delete.

Reply:
That comment wasn't as clear as I had hoped it would render.  I'm talking about the extra blank line below the derive.

Reply:
done